openwisp_monitoring/monitoring/static/monitoring/js/chart.js

```diff
@@ -16,6 +16,128 @@
         }
         return newArr;
     }
+
+    function adaptiveFilterPoints(charts, yRawVal) {
+
+        var convertToAdaptiveBytes = function(charts, type, i, multiplier, unit) {
+            charts[type].y[i] = Math.round((charts[type].y[i] * multiplier) * 100) / 100;
+            charts[type].hovertemplate[i] = charts[type].y[i] + ' ' + unit;
+        };
```

Comment:
You can improve this function which will also improve the readability. This is an utility function. This does not need to know anything about the chart because it does not perform any operation on it. I would update the signature of this function in following way.  


```js
var convertToAdaptiveBytes = function(value, multiplier) {
    return  Math.round((value * multiplier) * 100) / 100;
}
```
And you can do assignments on `charts[type].y[i]` and `charts[type].hovertemplate[i]` in the loop that calls this function below. 


@yashikajotwani12 work on this suggestion at the very last, because it does not affect the functionality anyway. Also, wait for green flag from @nemesisdesign on this suggestion before proceeding.

Reply:
we'd need to rename this `getAdaptiveBytes`.